extension/intents/window/window.js

```diff
@@ -0,0 +1,42 @@
+import * as intentRunner from "../../background/intentRunner.js";
+
+function findTargetWindowId(windowArray, currentWindowId, direction) {
+    const len = windowArray.length;
+    // find currentWindowId postion in array
+    const currentWindowIndex = windowArray.findIndex((window) => (window.id === currentWindowId));
+    let targetIndex = 0;
+    if ( direction === "next" ) {
+      targetIndex = Math.floor((currentWindowIndex + 1) % len);
+    } else {
+      targetIndex = Math.floor((currentWindowIndex - 1 + len) % len);
```

Comment:
`Math.floor` won't be necessary here, everything is an integer so it should stay an integer.